### ISSUE TITLE
How to transfer weights without creating new instance of YOLO?

### ISSUE THREAD

**Issue by hardikdava** [ultralytics/ultralytics#3664]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello, I am working on federated learning where I need to set weights i.e. `model.state_dict()` from another model. Is there anyway to do so in training mode?

### Additional

```
model_a = YOLO(...)
model_b = YOLO(...)

model_a_weights = get_weights(model_a)

Tranfer weights of model-a to model-b and get loose of model-b weights
model_b = set_weights(model_a_weights )
```
